x-pack/legacy/plugins/apm/server/routes/settings.ts

```diff
@@ -20,13 +20,19 @@ import { transactionSampleRateRt } from '../../common/runtime_types/transaction_
 // get list of configurations
 export const agentConfigurationRoute = createRoute(core => ({
   path: '/api/apm/settings/agent-configuration',
-  handler: async req => {
-    await createApmAgentConfigurationIndex(core.http.server);
+  handler: async (req, _, h) => {
+    try {
+      await createApmAgentConfigurationIndex(core.http.server);
 
-    const setup = await setupRequest(req);
-    return await listConfigurations({
-      setup
-    });
+      const setup = await setupRequest(req);
+      return await listConfigurations({ setup });
+    } catch (error) {
+      if (error.statusCode === 403) {
+        // return an empty list if user is not authorized
```

Comment:
I think you should only do this for `createApmAgentConfigurationIndex`. I don't think we should swallow errors for `listConfigurations` this way.

Reply:
This assumes that the apm_user as a minimum has read access to the index.

Reply:
The latest commit removed all the assumptions and displays a message to the user that a failure occurred when loading settings.

Index creation uses `callWithInternalUser` during plugin setup, so any index creation failures should not occur when a user loads the settings screen. but it will error if the user does not have permission to read from the .apm-agent-configuration index. as per your list of use cases above.

When testing with a user with the role `apm_user`, i discovered that it does not have read permissions on this index. Since it's not appropriate to call `callWithInternalUser` for querying this index, it will show the failure message. The alternative is to give the `apm_user` role the permission to read the agent config index. I created a PR to update the role, however it looks like it will not get approval (https://github.com/elastic/elasticsearch/pull/46423).